src/react_native/utils.cljs

```diff
@@ -0,0 +1,20 @@
+(ns react-native.utils)
+
+(defonce throttle (atom {}))
+
+(defn throttled-on-press
+  [{:keys [on-press allow-multiple-presses? throttle-duration]}]
+  (let [pressable-id (rand 10000)]
+    (if allow-multiple-presses?
+      on-press
+      (fn []
+        (when-not (get @throttle pressable-id)
+          (swap! throttle assoc pressable-id true)
+          (on-press)
+          (js/setTimeout
+           #(swap! throttle assoc pressable-id false)
```

Comment:
Just a slight improvement: we can check `allow-multiple-presses?` first, and if it's true we don't need to call `rand`.